bridge/src/main/java/com/bridge/renderHandler/repository/SpriteRepository.java

```diff
@@ -21,7 +22,14 @@ public SpriteRepository() {
      */
     @Override
     public void add(Sprite sprite) {
-        sprites.add(sprite);
+        sprites.add(sprite.getZ_index(), sprite);
```

Comment:
>  Shifts the element currently at that position (if any) and any subsequent elements to the right (adds one to their indices).
-- [source](https://docs.oracle.com/en/java/javase/17/docs//api/java.base/java/util/List.html#add(int,E))

I think this add method would shift sprites from their layers. What about using the retrieve method instead, to return a list ordered by its z index?

Reply:
What I meant was to add sprites to the list without the specific index.
`spites.add(sprite)`